spec/support/features/session_helper.rb

```diff
@@ -162,7 +162,7 @@ def acknowledge_and_confirm_recovery_code
       click_on button_text, class: 'recovery-code-continue'
 
       code_words.size.times do |index|
-        fill_in "recovery-#{index}", with: code_words[index].downcase
+        fill_in "recovery-#{index}", with: code_words[index].downcase + 'xxxx'
```

Comment:
Sorry, perhaps I misunderstood. I thought you were asking if we checked the length when we were confirming the existing password, not creating it. In that case, the user is creating the password, so before we allow that, we validate the length.

In the case of recovery code, the server creates it, so we know it is always the proper length. We don't do server-side validation of the recovery code length before we confirm that it matches what we have stored. That would be a waste of cycles IMO.